### ISSUE TITLE
Failure to detonate k8s.privilege-escalation.nodes-proxy

### ISSUE THREAD

**Issue by prsimoes** [DataDog/stratus-red-team#428]
**What is not working?**
The attack technique `k8s.privilege-escalation.nodes-proxy` is failing to fully detonate.

**What OS are you using?**
Mac OS X

**What is your Stratus Red Team version?**
`2.9.0`

**Full output?**
```
stratus detonate k8s.privilege-escalation.nodes-proxy
2023/11/16 16:46:01 Checking your authentication against kubernetes
2023/11/16 16:46:02 Creating /Users/user1/.stratus-red-team as it doesn't exist yet
2023/11/16 16:46:02 Installing Terraform in /Users/user1/.stratus-red-team/terraform
2023/11/16 16:46:05 Warming up k8s.privilege-escalation.nodes-proxy
2023/11/16 16:46:05 Initializing Terraform to spin up technique prerequisites
2023/11/16 16:46:10 Applying Terraform to spin up technique prerequisites
2023/11/16 16:46:43 Error during warm up. Cleaning up technique prerequisites with terraform destroy
2023/11/16 16:46:54 unable to run terraform apply on prerequisite: unable to apply Terraform: exit status 1

Error: Waiting for default secret of "stratus-red-team-np-name-xmpomdag/stratus-red-team-np-sa" to appear

  with kubernetes_service_account.sa,
  on main.tf line 51, in resource "kubernetes_service_account" "sa":
  51: resource "kubernetes_service_account" "sa" {
```

**Files in $HOME/.stratus-red-team?**
```
ls -lahR
total 140232
drwxr--r--    4 user1  staff   128B Nov 16 16:46 .
drwxr-x---+ 128 user1  staff   4.0K Nov 16 16:46 ..
drwxr--r--    8 user1  staff   256B Nov 16 16:46 k8s.privilege-escalation.nodes-proxy
-rwx------    1 user1  staff    68M Nov 16 16:46 terraform

./k8s.privilege-escalation.nodes-proxy:
total 40
drwxr--r--  8 user1  staff   256B Nov 16 16:46 .
drwxr--r--  4 user1  staff   128B Nov 16 16:46 ..
drwxr-xr-x  3 user1  staff    96B Nov 16 16:46 .terraform
-rw-r--r--  1 user1  staff     0B Nov 16 16:46 .terraform-initialized
-rw-r--r--  1 user1  staff   2.1K Nov 16 16:46 .terraform.lock.hcl
-rw-r--r--  1 user1  staff   2.2K Nov 16 16:46 main.tf
-rw-r--r--  1 user1  staff   156B Nov 16 16:46 terraform.tfstate
-rw-r--r--  1 user1  staff   4.8K Nov 16 16:46 terraform.tfstate.backup

./k8s.privilege-escalation.nodes-proxy/.terraform:
total 0
drwxr-xr-x  3 user1  staff    96B Nov 16 16:46 .
drwxr--r--  8 user1  staff   256B Nov 16 16:46 ..
drwxr-xr-x  3 user1  staff    96B Nov 16 16:46 providers

./k8s.privilege-escalation.nodes-proxy/.terraform/providers:
total 0
drwxr-xr-x  3 user1  staff    96B Nov 16 16:46 .
drwxr-xr-x  3 user1  staff    96B Nov 16 16:46 ..
drwxr-xr-x  3 user1  staff    96B Nov 16 16:46 registry.terraform.io

./k8s.privilege-escalation.nodes-proxy/.terraform/providers/registry.terraform.io:
total 0
drwxr-xr-x  3 user1  staff    96B Nov 16 16:46 .
drwxr-xr-x  3 user1  staff    96B Nov 16 16:46 ..
drwxr-xr-x  4 user1  staff   128B Nov 16 16:46 hashicorp

./k8s.privilege-escalation.nodes-proxy/.terraform/providers/registry.terraform.io/hashicorp:
total 0
drwxr-xr-x  4 user1  staff   128B Nov 16 16:46 .
drwxr-xr-x  3 user1  staff    96B Nov 16 16:46 ..
drwxr-xr-x  3 user1  staff    96B Nov 16 16:46 kubernetes
drwxr-xr-x  3 user1  staff    96B Nov 16 16:46 random

./k8s.privilege-escalation.nodes-proxy/.terraform/providers/registry.terraform.io/hashicorp/kubernetes:
total 0
drwxr-xr-x  3 user1  staff    96B Nov 16 16:46 .
drwxr-xr-x  4 user1  staff   128B Nov 16 16:46 ..
drwxr-xr-x  3 user1  staff    96B Nov 16 16:46 2.7.1

./k8s.privilege-escalation.nodes-proxy/.terraform/providers/registry.terraform.io/hashicorp/kubernetes/2.7.1:
total 0
drwxr-xr-x  3 user1  staff    96B Nov 16 16:46 .
drwxr-xr-x  3 user1  staff    96B Nov 16 16:46 ..
drwxr-xr-x  3 user1  staff    96B Nov 16 16:46 darwin_arm64

./k8s.privilege-escalation.nodes-proxy/.terraform/providers/registry.terraform.io/hashicorp/kubernetes/2.7.1/darwin_arm64:
total 108664
drwxr-xr-x  3 user1  staff    96B Nov 16 16:46 .
drwxr-xr-x  3 user1  staff    96B Nov 16 16:46 ..
-rwxr-xr-x  1 user1  staff    53M Nov 16 16:46 terraform-provider-kubernetes_v2.7.1_x5

./k8s.privilege-escalation.nodes-proxy/.terraform/providers/registry.terraform.io/hashicorp/random:
total 0
drwxr-xr-x  3 user1  staff    96B Nov 16 16:46 .
drwxr-xr-x  4 user1  staff   128B Nov 16 16:46 ..
drwxr-xr-x  3 user1  staff    96B Nov 16 16:46 3.5.1

./k8s.privilege-escalation.nodes-proxy/.terraform/providers/registry.terraform.io/hashicorp/random/3.5.1:
total 0
drwxr-xr-x  3 user1  staff    96B Nov 16 16:46 .
drwxr-xr-x  3 user1  staff    96B Nov 16 16:46 ..
drwxr-xr-x  3 user1  staff    96B Nov 16 16:46 darwin_arm64

./k8s.privilege-escalation.nodes-proxy/.terraform/providers/registry.terraform.io/hashicorp/random/3.5.1/darwin_arm64:
total 27880
drwxr-xr-x  3 user1  staff    96B Nov 16 16:46 .
drwxr-xr-x  3 user1  staff    96B Nov 16 16:46 ..
-rwxr-xr-x  1 user1  staff    14M Nov 16 16:46 terraform-provider-random_v3.5.1_x5
```

**Comment by prsimoes** [comment]
I'm using an EKS cluster, version 1.25.

**Comment by christophetd** [comment]
Was able to reproduce on EKS, will have a look shortly. Thanks for reporting!

**Comment by christophetd** [comment]
Appears to be related to https://github.com/hashicorp/terraform-provider-kubernetes/issues/1724

**Comment by christophetd** [comment]
Likely fixed by https://github.com/hashicorp/terraform-provider-kubernetes/pull/1634

**Comment by christophetd** [comment]
Will be released shortly under v2.10.1
